src/Api/Controllers/AccountsController.cs

```diff
@@ -194,6 +195,30 @@ public async Task PostPassword([FromBody]PasswordRequestModel model)
             await Task.Delay(2000);
             throw new BadRequestException(ModelState);
         }
+        
+        [HttpPost("set-password")]
+        public async Task SetPasswordAsync([FromBody]SetPasswordRequestModel model)
+        {
+            var user = await _userService.GetUserByPrincipalAsync(User);
+            if (user == null)
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+            var result = await _userService.SetPasswordAsync(user, model.NewMasterPasswordHash, model.Key);
+            if (result.Succeeded)
+            {
+                return;
+            }
+
+            foreach (var error in result.Errors)
+            {
+                ModelState.AddModelError(string.Empty, error.Description);
+            }
+
+            await Task.Delay(2000);
```

Comment:
I'm pretty sure we decided to remove this particular delay

Reply:
Done